code-studio/src/js/components/progress/course_progress.jsx

```diff
@@ -11,26 +11,33 @@ var CourseProgress = React.createClass({
     stages: React.PropTypes.arrayOf(React.PropTypes.shape({
       name: React.PropTypes.string,
       lesson_plan_html_url: React.PropTypes.string,
+      flex_category: React.PropTypes.string,
       levels: STAGE_PROGRESS_TYPE
     }))
   },
 
   render: function () {
-    var rows = this.props.stages.map(function (stage) {
-      return (
-        <div className='game-group' key={stage.name}>
-          <div className='stage'>
-            {stage.title}
-            <div className='stage-lesson-plan-link' style={{display: 'none'}}>
-              <a target='_blank' href={stage.lesson_plan_html_url}>
-                {dashboard.i18n.t('view_lesson_plan')}
-              </a>
+    var rows = [], stages = this.props.stages;
+    for (var i = 0; i < stages.length; ) {
+
+      if (stages[i].flex_category) {
+        var flexRows = [], previous = stages[i].flex_category;
+        for ( ; i < stages.length && stages[i].flex_category === previous; i++) {
```

Comment:
it'd be nice to get a comment explaining what's going on here; non-standard use of for loops can be an easy point of confusion for anyone who's just glancing over the code.